ui/lib/sync/addon/components/secrets/page/destinations.hbs

```diff
@@ -12,4 +12,114 @@
       <LinkTo @route="secrets.destinations" data-test-tab="Destinations">Destinations</LinkTo>
     </ul>
   </nav>
-</div>
\ No newline at end of file
+</div>
+
+<Toolbar>
+  <ToolbarFilters>
+    <SearchSelect
+      @options={{this.destinationTypes}}
+      @objectKeys={{array "id" "name"}}
+      @passObject={{true}}
+      @selectLimit={{1}}
+      @disallowNewItems={{true}}
+      @placeholder="Filter by type"
+      @inputValue={{if this.typeFilter (array this.typeFilter)}}
+      @onChange={{fn this.onFilterChange "type"}}
+      class="is-marginless"
+      data-test-filter="type"
+    />
+    <SearchSelect
+      @options={{this.destinationNames}}
+      @objectKeys={{array "id" "name"}}
+      @passObject={{true}}
+      @selectLimit={{1}}
+      @disallowNewItems={{true}}
+      @placeholder="Filter by name"
+      @inputValue={{if @nameFilter (array @nameFilter)}}
+      @onChange={{fn this.onFilterChange "name"}}
+      class="is-marginless has-left-padding-s"
+      data-test-filter="name"
+    />
+  </ToolbarFilters>
+  <ToolbarActions>
+    <ToolbarLink @route="secrets.destinations.create" @type="add" data-test-create-destination>
+      Create new destination
+    </ToolbarLink>
+  </ToolbarActions>
+</Toolbar>
+
+{{#if @destinations.meta.filteredTotal}}
+  <div class="has-bottom-margin-s">
+    {{#each @destinations as |destination index|}}
+      <ListItem
+        @linkPrefix={{this.mountPoint}}
+        @linkParams={{array "secrets.destinations.destination.details" destination.type destination.name}}
+        as |Item|
+      >
+        <Item.content>
+          <div>
+            <Icon @name={{destination.icon}} data-test-destination-icon={{index}} />
+            <span data-test-destination-name={{index}}>
+              {{destination.name}}
+            </span>
+          </div>
+          <code class="has-text-grey is-size-8" data-test-destination-type={{index}}>
+            {{destination.typeDisplayName}}
+          </code>
+        </Item.content>
+
+        <Item.menu as |Menu|>
+          {{#if destination.destinationPath.isLoading}}
+            <li class="action">
+              <Hds::Button disabled @color="tertiary" @icon="loading" @text="loading" @isIconOnly={{true}} />
```

Comment:
Chelsea made a new loading component specifically for dropdowns 
https://github.com/hashicorp/vault/blob/main/ui/app/templates/vault/cluster/policies/index.hbs#L99-L101

```
<li class="action">
 <LoadingDropdownOption />
</li>

Reply:
That's handy I will update!

Reply:
Dang this is in the app components not core addon 😞. I'll go ahead and move for wider use.

Reply:
Oh dang it!